docs/source/calling-smart-contracts.md

```diff
@@ -0,0 +1,125 @@
+smart contracts.
+
+## with Interface
+Use the `contract` decorator from `boa3.builtin.compile_time` using the script hash and create a class that have the 
+same methods you want call.
+```python
+# calling_with_interface.py
+from boa3.builtin.compile_time import public, contract
+
+@public
+def calling_other_contract() -> str:
+    greetings = HelloStrangerContract.hello_stranger('John Doe')
+    return greetings
+
+
+@contract('0x000102030405060708090A0B0C0D0E0F10111213')
+class HelloStrangerContract:
+    @staticmethod
+    def hello_stranger(name: str) -> str:
+        pass
+
+```
+
+### Calling native contracts
+Neo3-boa already has interfaces for all the [native contracts](https://docs.neo.org/docs/en-us/reference/scapi/framework/native.html) 
+that you can import from `boa3.builtin.nativecontract`
+```python
+# calling_native_contract.py
+from boa3.builtin.compile_time import public
+from boa3.builtin.nativecontract import NEO
+
+@public
+def calling_other_contract() -> str:
+    neo_symbol = NEO.symbol()
+    return neo_symbol
+```
+
+### Automate with CPM
+Instead of manually writing the smart contract interface, you can use [CPM](https://github.com/CityOfZion/cpm/tree/master#readme) 
+to generate it automatically. After installing Neo3-boa, you can install CPM by typing `install_cpm` on CLI (without the 
+`neo3-boa` prefix). Then, you'll need to create a [cpm.yaml config file](https://github.com/CityOfZion/cpm/blob/master/docs/config.md) 
+and put the smart contract information there, and [run cpm](https://github.com/CityOfZion/cpm#example-commands).
+
+For example, if you use CPM to create a [dice smart contract](https://dora.coz.io/contract/neo3/mainnet/0x4380f2c1de98bb267d3ea821897ec571a04fe3e0)
+interface, the following file will be generated:
+```python
+# cpm_out/python/dice/contract.py
+from boa3.builtin.type import UInt160, UInt256, ECPoint
+from boa3.builtin.compile_time import contract, display_name
+from typing import cast, Any
+
+
+@contract('0x4380f2c1de98bb267d3ea821897ec571a04fe3e0')
+class Dice:
+
+    @staticmethod
+    def rand_between(start: int, end: int) -> int: 
+        pass
+
+    @staticmethod
+    def map_bytes_onto_range(start: int, end: int, entropy: bytes) -> int: 
+        pass
+
+    @staticmethod
+    def roll_die(die: str) -> int: 
+        pass
+
+    @staticmethod
+    def roll_dice_with_entropy(die: str, precision: int, entropy: bytes) -> list: 
+        pass
+
+    @staticmethod
+    def update(script: bytes, manifest: bytes, data: Any) -> None: 
+        pass
+```
+
+Then, all you need to do is import this class onto your smart contract.
+```python
+# calling_with_cpm.py
+from boa3.builtin.compile_time import public
+from cpm_out.python.dice.contract import Dice
+
+
+@public
+def calling_other_contract() -> str:
+    d6_roll = Dice.rand_between(1, 6)
+    return "Die result is " + str(d6_roll)
```

Comment:
```suggestion
    return "Dice result is " + str(d6_roll)
```
also it should be good to use interop and native contract methods in the examples to better example how Neo developing works (not mandatory in this PR)
```suggestion
    return "Dice result is " + StdLib.itoa(d6_roll)
```